### PR TITLE
NF: Remove unused methods

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Code cleaning. Removing most of the methods which are never used, most of the members which are never read/setted, variables which are never read.

(Note: when `type var = expr`; occured and `var` was never read, I kept `expr` when it might have side effect.)

This was done thanks to Andrdoid Studio help. Commits are not always atomic because sometime it made sense to delete a bunch of stuff together. I tried to keep it as atomic as possible so that it is easier to bisect if it were required.
 
Note that I checked each commit manually. For each method deleted, I used Android Studio to check whether the method was indeed not called anywhere. For each member, I checked with android studio to see all places where it was read/assigned.

The good news is that if you decide to wait to merge it, there should be little merge conflict, since it only touch codes that is actually never used in AnkiDroid. And if there is a merge conflict it probably means that someone eventually found a reason to use this code, so it should be removed from this PR deletion.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/exception/NoSuchDeckException.java [hunk 1]
```diff
@@ -6,14 +6,6 @@
...
-    private final long mDeckId;
-
-    public NoSuchDeckException(long deckId) {
-        this.mDeckId = deckId;
-    }
-
-    /** The ID of the accessed deck */
-    public long getDeckId() {
-        return mDeckId;
+    public NoSuchDeckException() {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/exception/NoSuchDeckException.java:9]: Codacy Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574097)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 21
- Added 2
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.java  -6
+ AnkiDroid/src/main/java/com/ichi2/anim/ViewAnimation.java  -5
+ AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java  -3
+ AnkiDroid/src/main/java/com/ichi2/libanki/Tags.java  -4
+ AnkiDroid/src/main/java/com/ichi2/anki/AnkiFont.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574097&bid=18274928)

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.java [hunk 2]
```diff
@@ -274,79 +269,14 @@ private void showPickTranslationDialog() {
...
-            }
+        mPossibleTranslations = new ArrayList<>();
 
+        if (!mSource.toLowerCase(Locale.getDefault()).contentEquals(mSource)) {
+            showToastLong(gtxt(R.string.multimedia_editor_word_search_try_lower_case));
         }
 
-        return res;
+        returnFailure(getText(R.string.multimedia_editor_error_word_not_found).toString());
+        return;
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.java:279]: Codacy Issue found: [Avoid unnecessary return statements](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574097)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 21
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java  -3
+ AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.java  -6
+ AnkiDroid/src/main/java/com/ichi2/libanki/Tags.java  -4
+ AnkiDroid/src/main/java/com/ichi2/anki/AnkiFont.java  -2
+ AnkiDroid/src/main/java/com/ichi2/anim/ViewAnimation.java  -5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574097&bid=18274928)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 21
           

Complexity decreasing per file
==============================
+ AnkiDroid/src/main/java/com/ichi2/libanki/Tags.java  -4
+ AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/TranslationActivity.java  -6
+ AnkiDroid/src/main/java/com/ichi2/anki/Lookup.java  -3
+ AnkiDroid/src/main/java/com/ichi2/anim/ViewAnimation.java  -5
+ AnkiDroid/src/main/java/com/ichi2/anki/AnkiFont.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5574097&bid=18274928)

File: AnkiDroid/src/main/java/com/ichi2/ui/SlidingTabLayout.java [hunk 3]
```diff
@@ -98,17 +95,6 @@ public SlidingTabLayout(Context context, AttributeSet attrs, int defStyle) {
     }
 
  
-    /**
-     * Set the custom layout to be inflated for the tab views.
-     *
-     * @param layoutResId Layout id to be inflated
-     * @param textViewId id of the {@link TextView} in the inflated view
-     */
-    public void setCustomTabView(int layoutResId, int textViewId) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/ui/SlidingTabLayout.java:147]: Are you sure the CardTemplateEditor is working correctly after all these SlidingTabLayout changes?

**Pull request closed without merging**